### PR TITLE
add pallet_identity to the runtime

### PR DESCRIPTION
**Pull request by cuteolaf**

### CODE REVIEW

File: node/src/chain_spec.rs [hunk 1]
```diff
@@ -156,9 +157,13 @@ fn testnet_genesis(
...
 		},
 		transaction_payment: Default::default(),
-		
+
 		role_module: RoleModuleConfig {
 			new_admin: Some(get_account_id_from_seed::<sr25519::Public>("Alice")),
+			representatives: vec![
+				get_account_id_from_seed::<sr25519::Public>("Alice//Representative"),
+				get_account_id_from_seed::<sr25519::Public>("Bob//Representative"),
+			],
```
Comment (ilhanu) [inline node/src/chain_spec.rs:166]: Can we please pick someone else, as these keys already have a role to play in council etc. 
You can add new keys to the chain and give them a balance.

File: runtime/src/lib.rs [hunk 2]
```diff
@@ -43,24 +43,24 @@ use frame_system::{
...
-pub use pallet_nft::{self, Acc, CollectionId, ItemId, NftPermission};
 
+pub use pallet_asset_management;
+pub use pallet_bidding;
+pub use pallet_housing_fund;
+pub use pallet_onboarding;
 /// Import the template pallet.
 pub use pallet_roles;
-pub use pallet_housing_fund;
+pub use pallet_share_distributor;
```
Comment (ilhanu) [inline runtime/src/lib.rs:61]: structure is good 🔢

File: runtime/src/lib.rs [hunk 3]
```diff
@@ -328,14 +325,38 @@ impl pallet_uniques::Config for Runtime {
 	type Locker = ();
 }
 
+parameter_types! {
+	pub const BasicDeposit: Balance = deposit(1, 258); // 258 bytes on-chain
+	pub const FieldDeposit: Balance = deposit(0, 66); // 66 bytes on-chain
+	pub const SubAccountDeposit: Balance = deposit(1, 53);
+	pub const MaxAdditionalFields: u32 = 100;
+	pub const MaxRegistrars: u32 = 20;
```
Comment (ilhanu) [inline runtime/src/lib.rs:333]:
```suggestion
	pub const BasicDeposit: Balance = deposit(1, 258); // 258 bytes on-chain
	pub const FieldDeposit: Balance = deposit(0, 66); // 66 bytes on-chain
	pub const SubAccountDeposit: Balance = deposit(1, 53);
	pub const MaxAdditionalFields: u32 = 100;
	pub const MaxRegistrars: u32 = 1;
```
not sure about the despitfees per bytes how to read this ?

File: pallets/bidding/src/tests.rs [hunk 4]
```diff
@@ -702,10 +702,7 @@ fn process_onboarded_assets_should_succeed() {
 
 		events.pop();
 
-		let event = events
-			.pop()
-			.expect("Expected at least one EventRecord to be found")
-			.event;
+		let event = events.pop().expect("Expected at least one EventRecord to be found").event;
```
Comment (ilhanu) [inline pallets/bidding/src/tests.rs:705]: What does this change have to do with the identity pallet ?

**Review by ilhanu — CHANGES_REQUESTED** [review]
small changes and let's keep small fixes away from PR for what they are intended for.

File: runtime/src/lib.rs (hunk 3 above)
Comment (cuteolaf) [reply runtime/src/lib.rs:333]: I saw two example configurations.

https://github.com/paritytech/substrate/blob/a7ba55d3c54b9957c242f659e02f5b5a0f47b3ff/bin/node/runtime/src/lib.rs#L1312-L1314

https://github.com/PureStake/moonbeam/blob/2be5c91596e6a5920aabb587a24fa86b55c9cfd2/runtime/moonbeam/src/lib.rs#L653-L657

They look similar but I'm not sure what's the main difference. Still thinking about it.

File: pallets/bidding/src/tests.rs (hunk 4 above)
Comment (cuteolaf) [reply pallets/bidding/src/tests.rs:705]: Nothing. Just a formatting problem.

File: node/src/chain_spec.rs [hunk 5]
```diff
@@ -156,9 +161,13 @@ fn testnet_genesis(
 			key: Some(root_key),
 		},
 		transaction_payment: Default::default(),
-		
+
 		role_module: RoleModuleConfig {
 			new_admin: Some(get_account_id_from_seed::<sr25519::Public>("Alice")),
+			representatives: vec![
+				AccountId32::from_str("5HErWv2YuCrjoW3bDjGoist4vVryCAhWGk2YcCqdwTG1SMMH").unwrap(),
+				AccountId32::from_str("5FFDJcwqRCFPFKM44DhFC3ituGwgaQKxpp75Ger9xNs3GCiU").unwrap(),
```
Comment (ilhanu) [inline node/src/chain_spec.rs:169]: Which accounts are these ?

File: runtime/src/lib.rs [hunk 6]
```diff
@@ -328,14 +325,38 @@ impl pallet_uniques::Config for Runtime {
 	type Locker = ();
 }
 
+parameter_types! {
+	pub const BasicDeposit: Balance = deposit(1, 258); // 258 bytes on-chain
+	pub const FieldDeposit: Balance = deposit(0, 66); // 66 bytes on-chain
```
Comment (ilhanu) [inline runtime/src/lib.rs:330]: I think these can be stated as [contants](https://github.com/Fair-Squares/fair-squares/blob/main/runtime/src/constants.rs#L24-L26) as the ParityTech example you linked to us

**Review by ilhanu — COMMENTED** [review]
Regarding setting the identities, can you check [this](https://github.com/paritytech/substrate/blob/8161ee1aa1ccd122150e40a8aef23df3d0a16d02/frame/support/src/traits/hooks.rs#L108-L119) 

Apparently we can set stuff with frame trait `on_genesis` think this will be a root call.

File: node/src/chain_spec.rs (hunk 5 above)
Comment (cuteolaf) [reply node/src/chain_spec.rs:169]: Random accounts generated by subkey

File: runtime/src/lib.rs (hunk 6 above)
Comment (cuteolaf) [reply runtime/src/lib.rs:330]: updated

File: node/src/chain_spec.rs (hunk 5 above)
Comment (letodunc) [reply node/src/chain_spec.rs:169]: Use `get_account_id_from_seed` to use named accounts. Those representatives will be selected in the workflow later.

**Review by letodunc — CHANGES_REQUESTED** [review]

**Review by letodunc — APPROVED** [review]

**Review by ilhanu — APPROVED** [review]
Looks good, thanks for correcting some of the reviewed parts. Got that you don't want to implement the on-genesis implementation in this PR. It's still worth looking into it will just save us a lot of work to have  a genesis config of that.

**Review by ndkazu — APPROVED** [review]

**Pull request merged**